### DISCUSSION TITLE
How to extract css file whose name is not hash

### DISCUSSION THREAD

**Discussion by iWun** [Anidetrix/rollup-plugin-styles#157] [Q&A] [answered]
`styles({
      mode: ["extract", "index.css"],
      use: [
        'css', 
        'less',
      ],
      less: {
        javascriptEnabled: true,
      },
    }),`

I want to extract a css file whose name is not hash.
Any help would be greatly appreciated

**Comment by wilkinsonjack1993** [comment, marked as answer]
Hi @iWun,

I just ran into this problem. You need to set the `assetFileNames` property in the rollup config output:

```
import styles from "rollup-plugin-styles";

export default {
  output: {
    assetFileNames: "[name][extname]",
  },
  plugins: [styles()],
};
```

By default, I think it is  "[name]-[hash][extname]", which is why you get the hash at the end of the filename. N.B. This will increase the chances of name clash in your file output.
